### PR TITLE
add TensorDataset for multiprocess DataLoader

### PR DESCRIPTION
**Pull request by heavengate**
### PR types
New features

### PR changes
APIs

### Describe
- add `paddle.io.TensorDataset` API
td

### CODE REVIEW

**Review by LielinJiang — DISMISSED** [review]

File: python/paddle/fluid/tests/unittests/test_multiprocess_dataloader_dataset.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# limitations under the License.
+
+from __future__ import division
+
+import os
+import sys
+import six
+import time
+import unittest
+import multiprocessing
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_multiprocess_dataloader_dataset.py]: remove un-used importing, multiprocessing, time, ...

File: python/paddle/fluid/tests/unittests/test_multiprocess_dataloader_dataset.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                places=place,
+                num_workers=num_workers,
+                batch_size=1,
+                drop_last=True)
+
+            for input, label in dataloader():
+                assert len(input) == 1
+                assert len(label) == 1
+                assert input.shape == [1, 3, 4]
+                assert label.shape == [1, 1]
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_multiprocess_dataloader_dataset.py:50]: you also can check the input/label type.  whether it is equal to input_np/label_np.

File: python/paddle/fluid/dataloader/dataset.py [hunk 3]
```diff
@@ -222,3 +222,52 @@ def __getitem__(self, idx):
     def __len__(self):
         raise RuntimeError("'{}' should not be called for IterableDataset" \
                 "{}".format('__len__', self.__class__.__name__))
+
+
+class TensorDataset(Dataset):
```
Comment (qingqing01) [inline python/paddle/fluid/dataloader/dataset.py:228]: can TensorDataset only be used in dygraph? if so,  raise error in static mode

Comment (heavengate) [reply python/paddle/fluid/dataloader/dataset.py:228]: Done, thanks!

File: python/paddle/fluid/tests/unittests/test_multiprocess_dataloader_dataset.py (hunk 1 above)
Comment (heavengate) [reply python/paddle/fluid/tests/unittests/test_multiprocess_dataloader_dataset.py]: Done, thanks!

File: python/paddle/fluid/tests/unittests/test_multiprocess_dataloader_dataset.py (hunk 2 above)
Comment (heavengate) [reply python/paddle/fluid/tests/unittests/test_multiprocess_dataloader_dataset.py:50]: Done, thanks!

**Review by Heeenrrry — APPROVED** [review]
LGTM

**Review by qingqing01 — APPROVED** [review]

**Review by chalsliu — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**